### ISSUE TITLE
Scale to big projects

### ISSUE THREAD

**Issue by duckie** [jessevdk/cldoc#77]
I'm trying to run cldoc on a quite big project and there is a big issue here. I have to generate all of the documentation in one call to the tool.

This is hard on resources for the machine, since cldoc loads everythong in the RAM. This is annoying for the developer, since any failure in the process means re-generate _everything_. The other issue is that the compiler options and flags often change from one directory to another.

What would be great would be to generate the xmls of subdirectories, then having a way from cldoc to concatenate those then generate the full web app.

What do you think ?

**Comment by jessevdk** [comment]
Yes, these situations are currently not really handled well with cldoc. There are some other things that would be nice, like only rebuilding docs for things that have actually changed since the last time (incremental build). Unfortunately this will require some significant changes. I don't see any immediate obstacles, apart from time :)
